### DISCUSSION TITLE
problema con el createPayment

### DISCUSSION THREAD

**Discussion by Fabianjofre** [BlackBoxVision/react-native-mercadopago-px#44] [Q&A]
Estimado, estamos usando esta librería en un proyecto y tengo un problema al invocar al cretatePayment. 
genero sin problema el preferenceId pero cuando llamo al createPayment me da el siguiente error.
       null is not an object (evaluating '_reactNative.NativeModules.ReactNativeMercadopagoPx.createPayment')

podrías orientarme en como solucionarlo?

Gracias,
Fabian.

**Comment by JonatanSalas** [comment]
Hola Fabian, paso a pedirte algunos detalles: 

- ¿Qué versión de React Native estás usando?
- ¿Qué versión de nuestra librería estás usando?
- ¿En qué dispositivo se produce este error? ¿Es en iOS, Android o ambos?

**Reply by Fabianjofre** [reply]
Hola Jonatan,

estamos usando la versión 0.7.2, el error en Android, en IOS aun no lo probé.

    "react-native": "https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz",


Saludos,
Fabian.

**Reply by JonatanSalas** [reply]
Hola Fabian, por lo que veo están usando Expo para manejar el proyecto. 

Si estás usando el [managed-workflow](https://docs.expo.io/introduction/managed-vs-bare/#managed-workflow) de Expo no va a ser posible que corras el checkout porque básicamente se requiere una configuración a nivel código nativo, solo funcionamos con proyectos que vayan por el [bare-workflow](https://docs.expo.io/introduction/managed-vs-bare/#bare-workflow). 

Por lo que veo estás usando la SDK 38 de Expo, a partir de la versión 41 de Expo sacaron la feature [config plugins](https://docs.expo.io/guides/config-plugins) que permite básicamente que puedas mantener tu app con el managed-workflow y por atrás configurar el cliente de Expo. 

Nosotros aún no dimos soporte para Config Plugins pero lo tenemos en [Backlog](https://github.com/BlackBoxVision/react-native-mercadopago-px/issues/45) para que aquellos que usen Expo con SDK >= 41 puedan instalar nuestra librería y usarla sin problemas.

**Voy a agregar una sección en la documentación especificando que para que los proyectos con SDK < 41 de Expo funcionen tienen que ser proyectos eyectados o con el bare-workflow.**

**Reply by JonatanSalas** [reply]
@Fabianjofre ya deje la documentación tanto en inglés como en español respecto de Expo. Te paso el link a la documentación en Español: https://react-native-mercadopago-px.blackbox-vision.tech/es/docs/compatibility/
